modules/test/playwright/helpers/FeatureFlagApiHelper.ts

```diff
@@ -13,7 +13,6 @@ export class FeatureFlagApiHelper {
 	}
 
 	async updateFeatureFlag(key: string, enabled: boolean) {
-		await this.page.goto('/');
```

Comment:
I removed navigation here as it negates usage of page fixtures. It was unexpected to me, and needlessly worsened performance.

The downside it that there needs to be at least one navigation before `updateFeatureFlag` can be called. If not, we get a cryptic error:
![image](https://github.com/liferay-frontend/liferay-portal/assets/5435511/57e5a438-abc1-434c-9c1a-1c664882f340)

This is why I added the order change in `objectsFolder.spec.ts`

Reply:
IMO FF should be fixtures so that each test declares which FFs it needs and they are automagically set on/off by our testing architecture.

Reply:
Oh, I saw just now that the FF is turned on/off with a fetch, which is kind-of headless....

Maybe we can simply navigate to the home page when the test starts (in a global setup) and, that way, guarantee that fetch is going to work no matter where the test is.

Reply:
> Maybe we can simply navigate to the home page when the test starts (in a global setup) and, that way, guarantee that fetch is going to work no matter where the test is.

Unfortunately, it doesn't seem to work that way. I expected home page would be "active" after login, but it's not. I am not sure why. Login state is preserved, but looks like not everything else related to navigation :thinking:

Reply:
Yeah, the current login is just loading the cookies into the context of each test but that doesn't mean the browser will have the `Liferay` object defined, obviously, because no HTML is being interpreted.

In other words, the current login is like opening a browser with prepopulated cookies (like when you click "remember me"), not like going to the home page.

Reply:
So, we could add a fixure that navigates to home page, and have it composed with all tests, but:
1. That will result in quite a lot of boilerplate.
2. It's still not clear if want navigation in fixtures, see https://github.com/liferay-frontend/liferay-portal/pull/3927#discussion_r1448705875. If we implement it like this, we would need to make sure nobody adds navigation in fixtures. If a dev does it, it may result in flaky behavior, as the order of navigation would be uncertain.

I don't see how to implement this in a way to keep logic under the hood. One way or another, devs will need to know how to work around this odd quirk.

Reply:
I've fixed it in my login_fixture PR, in fact, see this block of code -> https://github.com/liferay/liferay-portal/commit/68586db716a18e4ef2f05486c05bff8043a026c2#diff-0dc401334992febbdd325750039c0a124f97e88116f031f78b6c7eef0384aba6R58-R67

What I do is trying to invoke `fetch` and if it fails with `ReferenceError`, go to page `/` and retry.

Since the code is encapsulated in a fixture it's a "clean" solution, meaning that the tests don't need to bother with it.

I tried to do it in the setup but it's not possible, because setup only runs once and the navigation to / needs to be done for each `Page`.

Reply:
I know I'm navigating inside a fixture but in this case I think it shouldn't be a problem because I only navigate in case the browser is still empty (nothing has been loaded yet) and the only side effect will be that the test asking for the fixture finds the home page preloaded when it enters which should be inocuous.

Reply:
In other words, navigating inside a fixture is not a problem. The problem is assuming (from a test) that because you asked for a fixture, the BrowerContext will be in a defined location.

Tests should navigate to the proper initial place as their very first action.

Reply:
> I've fixed it in my login_fixture PR, in fact, see this block of code

Looks good, but I still have strong doubts abut performance. It's nothing specific to the fixture, but comes from the fact that, one way or another, we need to make a ton of requests, one per FF. I'd much prefer to see an endpoint that can handle a list of FFs and set them all at once. But, that's out of scope here. In the case of this PR, I think we can push as it as, with FFs within tests. Once the fixture approach is ready, we can refactor.

Reply:
>  I'd much prefer to see an endpoint that can handle a list of FFs and set them all at once.

That's definitely possible with the proposed code structure so I'm OK. I mean, I don't want to make it fast at this point, just make it evolvable, meaning we don't need to touch any test in the future if we want to optimize the code and, for example, turn all FFs in one single request, as you say.

Reply:
After thinking on this, I figured the performance issue is not as bad as I initially though. It was a reflex reaction to impact of requests on standard UI, with interaction between server and browser anywhere in the world. For tests, it's always a local request, so it should be reasonably fast. It is still a performance issue, but not a severe one, something we can push down in priority.

Reply:
I removed code changing how FFs are set from PR. We're using `updateFeatureFlag` util as it is, until fixure FF is available.